README.md

```diff
@@ -94,6 +94,483 @@ Try the workshop at https://apprunnerworkshop.com
 Read the docs at https://docs.aws.amazon.com/apprunner
 ```
 
+## Use Pulumi resources with CDK Constructs
+
+It is possible to use Pulumi and CDK resources side-by-side. In order to pass a
+Pulumi Output value into a CDK resource you can use the [asString][./api-docs/README.md#asString], [asList][./api-docs/README.md#asList], &
+[asNumber](./api-docs/README.md#asNumber) functions. Conversely, in order to pass a CDK attribute to a Pulumi
+resource, you can use the [Stack.asOutput](./api-docs/README.md#asOutput) function to convert the CDK resource
+to a Pulumi Output value.
+
+### Example
+
+```ts
+import * as pulumicdk from '@pulumi/cdk';
+import * as aws from '@pulumi/aws';
+import * as s3ObjectLambda from 'aws-cdk-lib/aws-s3objectlambda';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+
+const app = new pulumicdk.App('app', (scope: pulumicdk.App) => {
+    const stack = new pulumicdk.Stack('accesspoint-stack');
+    const bucket = new s3.Bucket(stack, 'example-bucket');
+
+    const policyDoc = new iam.PolicyDocument();
+    policyDoc.addStatements(...);
+
+    const ap = new aws.s3.AccessPoint('exampleBucketAP', {
+      // Use `asOutput` to convert the bucketName attribute to a Pulumi Output
+      bucket: stack.asOutput(bucket.bucketName),
+      name: S3_ACCESS_POINT_NAME,
+      policy: policyDoc.toJSON(),
+    }, { parent: scope });
+
+    const objectLambdaAP = new s3ObjectLambda.CfnAccessPoint(stack, 's3ObjectLambdaAP', {
+      name: OBJECT_LAMBDA_ACCESS_POINT_NAME,
+        objectLambdaConfiguration: {
+          // Use `asString` to convert a Pulumi Output to a string value
+          supportingAccessPoint: pulumicdk.asString(ap.arn),
+          transformationConfigurations: [...],
+        },
+    });
+});
+```
+
+### L2 Example
+
+You can also use `asString` in combination with CDK `fromXXX` methods to up cast
+to the L2.
+
+```ts
+const app = new pulumicdk.App('app', (scope: pulumicdk.App) => {
+    const stack = new pulumicdk.Stack('example-stack');
+    const zone = new aws.route53.Zone('example-zone', {
+      name: 'cooldomain.io',
+    });
+
+    const hostedZone = aws_route53.HostedZone.fromHostedZoneAttributes(
+      this,
+      'hosted-zone',
+      {
+        zoneName: asString(zone.name),
+        hostedZoneId: asString(zone.zoneId),
+      },
+    );
+
+    new aws_route53.CnameRecord(this, 'record', {
+      zone: hostedZone,
+      domainName: 'example.com',
+      recordName: 'test',
+    });
+});
+
+```
+
+## Create Pulumi outputs
+
+In order to create Pulumi [Stack outputs](https://www.pulumi.com/docs/iac/concepts/stacks/#outputs)
+you have to add the outputs to the [App outputs](./api-docs/README.md#AppOutputs). You can do this
```

Comment:
This is a little confusing because everything is called OUtputs. Perhaps. "You also need to propagate the app outputs out all the way to the Pulumi Stack outputs."